src/loaddb/load_db.c

```diff
@@ -1361,3 +1372,196 @@ load_object_file (load_args * args, int *exit_status)
   return split (args->periodic_commit, args->object_file, c_handler, b_handler);
 }
+
+static T_SCHEMA_FILE_LIST_INFO **
+ldr_check_file_list (std::string & file_name, int &num_files, int &error_code)
+{
+  FILE *schema_fp;
+  T_SCHEMA_FILE_LIST_INFO *schema_object_file = NULL;
+  T_SCHEMA_FILE_LIST_INFO **schema_info = NULL;
+  T_SCHEMA_FILE_LIST_INFO **new_schema_info = NULL;
+  char buffer[PATH_MAX] = { 0, };
+  std::string read_file_name = "";
+
+  error_code = NO_ERROR;
+
+  schema_fp = ldr_check_file (file_name, error_code);
+  if (error_code != NO_ERROR && schema_fp == NULL)
+    {
+      goto error_return;
+    }
+  else if (schema_fp == NULL)
+    {
+      return NULL;
+    }
+
+  while (fgets ((char *) buffer, LINE_MAX, schema_fp) != NULL)
+    {
+      trim (buffer);
+
+      if (buffer[0] == '\0')
+	{
+	  continue;
+	}
+
+      if (schema_info == NULL)
+	{
+	  schema_info = (T_SCHEMA_FILE_LIST_INFO **) malloc (sizeof (T_SCHEMA_FILE_LIST_INFO *));
+	  if (schema_info == NULL)
+	    {
+	      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1,
+		      sizeof (T_SCHEMA_FILE_LIST_INFO *));
+	      goto error_return;
+	    }
+	}
+      else
+	{
+	  new_schema_info =
+	    (T_SCHEMA_FILE_LIST_INFO **) realloc (schema_info, sizeof (T_SCHEMA_FILE_LIST_INFO *) * (num_files + 1));
+	  if (new_schema_info == NULL)
+	    {
+	      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1,
+		      sizeof (T_SCHEMA_FILE_LIST_INFO *) * (num_files + 1));
+	      goto error_return;
+	    }
+
+	  schema_info = new_schema_info;
+	}
+
+      schema_object_file = (T_SCHEMA_FILE_LIST_INFO *) malloc (sizeof (T_SCHEMA_FILE_LIST_INFO));
+      if (schema_object_file == NULL)
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, sizeof (T_SCHEMA_FILE_LIST_INFO));
+	  goto error_return;
+	}
+
+      schema_info[num_files] = schema_object_file;
+
+      num_files++;
+
+      strcpy (schema_object_file->schema_file_name, buffer);
+      read_file_name = buffer;
+      schema_object_file->schema_fp = ldr_check_file (read_file_name, error_code);
+      if (error_code != NO_ERROR && schema_object_file->schema_fp == NULL)
+	{
+	  goto error_return;
+	}
+    }
+
+  if (schema_info == NULL)
+    {
+      const char *msg_format =
+	msgcat_message (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_LOADDB, LOADDB_MSG_EMPTY_SCHEMA_FILE_LIST);
+      print_log_msg (1, msg_format, file_name.c_str ());
+      util_log_write_errstr (msg_format, file_name.c_str ());
+      goto error_return;
+    }
+
+  if (schema_fp != NULL)
+    {
+      fclose_and_init (schema_fp);
+    }
+
+  return schema_info;
+
+error_return:
+
+  if (schema_fp != NULL)
+    {
+      fclose_and_init (schema_fp);
+    }
+
+  ldr_free_and_fclose (schema_info, num_files);
+
+  error_code = ER_FAILED;
+  num_files = 0;
+  return NULL;
+}
+
+static void
+ldr_free_and_fclose (T_SCHEMA_FILE_LIST_INFO ** file_list, int num)
+{
+  int i;
```

Comment:
Not serious problem. 

Could we think of idx instead of i?
It seems so general name. Of course we are easy to know that is file list index.
But to me, It is seems so general. 

But it is up to you, it is very subjective idea.

Reply:
Thank you for your comments.
I will fixed it.